src/nomad_simulations/model_system.py

```diff
@@ -317,59 +562,79 @@ class AtomicCell(Cell):
-        ase_atoms.set_pbc(self.periodic_boundary_conditions)
-
-        # Positions (ensure they are parsed)
-        if self.positions is not None:
-            if len(self.positions) != len(self.atoms_state):
-                logger.error(
-                    'Length of `AtomicCell.positions` does not coincide with the length of the `AtomicCell.atoms_state`.'
-                )
-                return None
-            ase_atoms.set_positions(self.positions.to('angstrom').magnitude)
-        else:
-            logger.error('Could not find `AtomicCell.positions`.')
-            return None
-
-        # Lattice vectors
-        if self.lattice_vectors is not None:
-            ase_atoms.set_cell(self.lattice_vectors.to('angstrom').magnitude)
-        else:
-            logger.info('Could not find `AtomicCell.lattice_vectors`.')
+        self.ase_atoms = ase.Atoms(
+            symbols=[atom_state.chemical_symbol for atom_state in self.atoms_state],
+            pbc=self.periodic_boundary_conditions,
+            positions=self.positions.to('angstrom').magnitude,
+            cell=self.lattice_vectors.to('angstrom').magnitude,
+        )
+        return self
 
-        return ase_atoms
+    def __init__(self, m_def: Section = None, m_context: Context = None, **kwargs):
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+        self.analyze_geometry = kwargs.get('analyze_geometry', False)
+        self.ase_atoms = None
 
-    def normalize(self, archive, logger) -> None:
+    def normalize(self, archive, logger: BoundLogger):
+        """When `analyze_geometry` is `true`, perform the geometry analysis.
+        All intermediate artifacts are stored in the `AtomicCell` section.
+        Only `GeometryDistribution` sections are retained in the archive written to disk."""
         super().normalize(archive, logger)
+        if self.analyze_geometry:
+            # set up neighbor list
+            if not self.geometry_analysis_cutoffs:
+                self.geometry_analysis_cutoffs = ase_nl.natural_cutoffs(
+                    self.ase_atoms, mult=3.0
+                )
+            self.neighbor_list = ase_nl.build_neighbor_list(
+                self.ase_atoms,
+                self.geometry_analysis_cutoffs,
+                self_interaction=False,
+            )
 
-        # Set the name of the section
-        self.name = self.m_def.name if self.name is None else self.name
+            # set up distributions
+            self.simple_distributions = DistributionFactory(
+                self.ase_atoms,
+                self.neighbor_list,
+            ).produce_distributions()
+
+            self.histogram_distributions, self.distributions = [], []
+            for distribution in self.simple_distributions:
+                if distribution.type == 'distances':
+                    bins = np.arange(
+                        np.arange(0, max(self.geometry_analysis_cutoffs), 0.001)
```

Comment:
> The bin spacing (i.e., 0.001 in this case) should also be giving as a parameter, similar to the cutoff.

> I would probably specify a default number of bins, rather than a bin spacing

I see your point, however, these values will become searchable in the whole of NOMAD. That's part of the reason for generating a histogram (the other is storage capacity).

Anyhow, I'd stick with a single, default distribution. I chose 0.001 angstrom, as it's the limit of accuracy for any method. If you want, I can change the value, though.

Reply:
The bin spacing is set under the `geometry_distributions` subsection, so this is already covered.
`normalize` does not accept any `kwargs`, so it's not appropriate to modify the bin spacing there either.
This is what `DistributionFactory` is for.

Actually, on 2nd thought, I'll also move the cutoff under `GeometryDistribution`.